src/drivers/intel/hda/hda-dma.c

```diff
@@ -756,9 +759,9 @@ static int hda_dma_set_config(struct dma_chan_data *channel,
 	     config->src_width <= 2))
 		dgcs |= DGCS_SCS;
 
-	/* set DGCS.FIFORDY for output dma */
-	if ((config->cyclic && config->direction == DMA_DIR_MEM_TO_DEV) ||
-	    (!config->cyclic && config->direction == DMA_DIR_LMEM_TO_HMEM))
+	/* set DGCS.FIFORDY for host output dma */
+	if ((config->cyclic && config->direction == DMA_DIR_HMEM_TO_LMEM) ||
+	    (config->cyclic && config->direction == DMA_DIR_LMEM_TO_HMEM))
```

Comment:
I think the !config->cyclic should be treated as a separate patch since that part was always false.